### ISSUE TITLE
ipython broken for python 3.6 because of jedi 0.18.0

### ISSUE THREAD

**Issue by jrett2** [ipython/ipython#12821]
```
jrett@linux:~$ python3.6 -m venv /home/jrett/venv/qa36/
jrett@linux:~$ source /home/jrett/venv/qa36/bin/activate
(qa36) jrett@ts-prod:~$ pip install ipython
<snip>
Installing collected packages: pickleshare, wcwidth, prompt-toolkit, decorator, backcall, ipython-genutils, six, traitlets, ptyprocess, pexpect, pygments, parso, jedi, ipython
Successfully installed backcall-0.2.0 decorator-4.4.2 ipython-7.16.1 ipython-genutils-0.2.0 jedi-0.18.0 parso-0.8.1 pexpect-4.8.0 pickleshare-0.7.5 prompt-toolkit-3.0.16 ptyprocess-0.7.0 pygments-2.8.0 six-1.15.0 traitlets-4.3.3 wcwidth-0.2.5

(qa36) jrett@ts-prod:~$ pip freeze
backcall==0.2.0
decorator==4.4.2
ipython==7.16.1
ipython-genutils==0.2.0
jedi==0.18.0
parso==0.8.1
pexpect==4.8.0
pickleshare==0.7.5
pkg-resources==0.0.0
prompt-toolkit==3.0.16
ptyprocess==0.7.0
Pygments==2.8.0
six==1.15.0
traitlets==4.3.3
wcwidth==0.2.5

(qa36) jrett@ts-prod:~/src/qa$ ipython
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: ?
Traceback (most recent call last):
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/terminal/ptutils.py", line 113, in get_completions
    yield from self._get_completions(body, offset, cursor_position, self.ipy_completer)
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/terminal/ptutils.py", line 126, in _get_completions
    for c in completions:
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/core/completer.py", line 438, in _deduplicate_completions
    completions = list(completions)
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/core/completer.py", line 1818, in completions
    for c in self._completions(text, offset, _timeout=self.jedi_compute_type_timeout/1000):
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/core/completer.py", line 1862, in _completions
    full_text=full_text, cursor_line=cursor_line, cursor_pos=cursor_column)
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/core/completer.py", line 2030, in _complete
    cursor_pos, cursor_line, full_text)
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/IPython/core/completer.py", line 1374, in _jedi_matches
    text[:offset], namespaces, column=cursor_column, line=cursor_line + 1)
  File "/home/jrett/venv/qa36/lib/python3.6/site-packages/jedi/api/__init__.py", line 726, in __init__
    project=Project(Path.cwd()), **kwds)
TypeError: __init__() got an unexpected keyword argument 'column'

```

Obviously the issue is Jedi 0.18.0. ipython should release a 7.16.2 with a fixed requirements file:
https://github.com/ipython/ipython/commit/90fb12b53e52314979167098de1c47f75f8fb14e

**Comment by Carreau** [comment]
> Obviously the issue is Jedi 0.18.0. ipython should release a 7.16.2 with a fixed requirements file:
> [90fb12b](https://github.com/ipython/ipython/commit/90fb12b53e52314979167098de1c47f75f8fb14e)

That unfortunately would not work, if I release 7.16.2 that pin jedi; then pip will happily say "no I can't upgrade/install to 7.16.2" and install 7.16.1.

There is unfortunately not much I can do at this point beyond yanking all old releases.

**Comment by bmerry** [comment]
> That unfortunately would not work, if I release 7.16.2 that pin jedi; then pip will happily say "no I can't upgrade/install to 7.16.2" and install 7.16.1.

Out of interest, why is adding the pin the right solution for master (#12757) but the wrong solution for Python 3.6 support?

If a pin won't work (and it sounds like #12740 got rather bogged down in that issue), perhaps it would be useful to add a version/feature check after importing Jedi and if it's going to break, printing a warning and advice on how to fix it? Unfortunately not everyone's first reaction is "let me Google that", but rather "IPython is awful and just crashes all the time."

**Comment by MrMino** [comment]
@bmerry master is the next major (8.0) - it's not released yet

**Comment by bmerry** [comment]
Ah, I see #12757 was reverted anyway.

I had to do a bit of digging to figure out the situation. For anyone else following along: IPython 7.20 has a fix to make it compatible with Jedi 0.18 (#12793), but doesn't support Python 3.6. The change looks reasonably small, so maybe it's possible to backport it to 7.16? Obviously that's work for someone to do, and I assume the maintainers would have done it by now if they had unlimited free time, so presumably it'll fall onto someone who particularly wants things to work.

**Comment by MrMino** [comment]
@bialix as it was already mentioned by Carreau in lots of other threads, this would make pip downgrade IPython in some particular cases (I don't remember specifics). In any case, whatever else there is to discuss about this, it should be discussed in #13100.
